platform/ios/src/MGLMapView.mm

```diff
@@ -4125,6 +4125,11 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
                 return true;
             }
             
+            if ([self.delegate respondsToSelector:@selector(mapView:canSelectAnnotation:)] &&
+                ![self.delegate mapView:self canSelectAnnotation:annotation]) {
```

Comment:
Should this delegate method apply to all annotations or only shape annotations? Point annotations already have a `MGLAnnotationImage.enabled` and `MGLAnnotationView.enabled`, so a redundant option without a defined order of precedence could be confusing. The `enabled` properties can be more efficient because they’re declarative: application code doesn’t run when MGLMapView needs to check whether a point annotation is interactive.

Consider renaming the method `-mapView:shapeAnnotationIsEnabled:` and mentioning the `enabled` properties in its documentation. There’s precedent in MGLMapViewDelegate for shape annotation–specific options.